### PR TITLE
[GUI] Add Unlock/Relock context flow

### LINKED ISSUE #553: Add option to automatically relock wallet when unlocked
### Describe the issue

The feature that the wallet automatically relocks after certain actions never worked 100% and was removed some time ago.

Plan is to have relock fixed and make relocking optional.

### LINKED ISSUE #1035: [Backport][Model][GUI] Back port and connect contextual unlock/lock flow.
Right now we don't have the contextual unlock/relock flow fully implemented. As it's an abstract flow, can be back ported from bitcoin (walletModel side only) and connected to our UI.

Footnote: remember to add the unlockForStakingOnly in the flow.

### LINKED ISSUE #1171: [Feature request] Offer to type password immediately instead of returning error message
When clicking on "Create Cold Stake Address" on Staker wallet while wallet is Locked, error message is returned. This feature request is to open up window to type password immediately and re-locking the wallet after address is successfully created.

featureRequest21

### LINKED ISSUE #1365: [UX Enhancement]  Wallet unlock option upon hitting Send PIV, then re-lock to last state.
**Aim:** To be able to hit Send PIV button, get a popup unlock to send password prompt, and after entering password and hitting OK, for it to send PIVX, then automatically lock the wallet back to its last known state.  

**Why:** This would significantly improve the user experience and flow when sending coins using the PIVX core wallet. (which is an often used key function of the wallet)

**Steps Required Currently:** 
(requires **7** clicks, or 6 Clicks + Enter)
1. Click Send PIV
2. Get told to unlock wallet - Click OK
3. Click lock icon in status bar and select Unlock wallet.
4. Enter password, hit Enter or Click OK.
5. Click Send PIV again.
6. Get confirmation prompt. Click OK to finally send.
7. Click lock icon in status bar and Click Lock wallet to re-lock. (or Staking Only to stake)

**Proposed Change 1** - Original Qt wallet behavior
(requires **3** clicks, or 2 Clicks + Enter) 
1. Click Send PIV
2. If wallet is locked/staking-only, popup appears to enter password. Enter password, hit Enter or Click OK. 
3. Get confirmation prompt. Click OK. Coin gets sent and wallet automatically gets locked to previous status. (locked or staking only)

**Proposed Change 2** - Combined confirmation & unlock password prompt
(requires **2** clicks, or 1 Click + Enter) 
1. Click Send PIV
2. If wallet is locked/staking-only, confirmation prompt popup appears along with a field to enter password to send. Enter password, hit Enter or Click OK. Coin gets sent and wallet automatically gets locked to previous status. (locked or staking only)

Obviously, change 2 would be best, with the least clicks required to send. But even change 1 would be a significant reduction in required user interaction to just send some coins.

### PR DESCRIPTION
**Pull request by random-zebra**
Fix `UnlockContext` class and connect it where needed.
Now, once an action requires the wallet to be fully unlocked, the askpassphrase dialog is presented to the user (if the status is "Locked" or "UnlockedForStaking"). This way, the password can be entered directly, and the operation can resume without needing additional clicks by the user.
The previous locking status is restored when the action completes.

Interested flows:
- Send transaction (closes #1365)
- Create New Receive Address
- Create New Payment Request
- Create Masternode Controller
- Create Cold Staking Address (closes #1171)
- Send Cold Stake Delegation
- Wallet Data: Add multisend recipient
- Tools: Sign Message
- Tools: BIP38 tool
- Upgrade wallet to HD

Also
Closes #553
Closes #1035
And maybe #1152 becomes not needed now.

### CODE REVIEW

File: src/qt/pivx/coldstakingwidget.cpp [hunk 1]
```diff
@@ -382,16 +382,25 @@ void ColdStakingWidget::updateDisplayUnit() {
     }
 }
 
-void ColdStakingWidget::showList(bool show){
+void ColdStakingWidget::showList(bool show)
+{
```
Comment (Fuzzbawls) [inline src/qt/pivx/coldstakingwidget.cpp]: Again, would like to see arbitrary style changes in their own commit, same as my comment in #1389.

Comment (random-zebra) [reply src/qt/pivx/coldstakingwidget.cpp]: yup was already going to factor them out in their own commit here too, after reading your comment in the other PR :+1:

**Comment by random-zebra** [comment]
Styling consistency changes have been grouped in the first commit (and expanded to include all files edited).

**Comment by random-zebra** [comment]
Rebased.

**Comment by random-zebra** [comment]
Rebased

**Comment by random-zebra** [comment]
Rebased, and added unlock context for "upgrade to HD" flow.
Ready for review / testing.

**Comment by random-zebra** [comment]
Commits squashed as per request.

**Comment by furszy** [comment]
Needs rebase.

**Comment by random-zebra** [comment]
Rebased... again.

**Pull request merged**